### PR TITLE
Makefile and test.sh updates

### PR DESCRIPTION
**Pull request by pdietl**
Discarded the useless make commands which were replaced by the test.sh script.

Added better makefile rules to clean up the makefile.
Changed test.sh to use better Bash syntax and to be executable by default

### CODE REVIEW

File: makefile [hunk 1]
```diff
@@ -1,88 +1,30 @@
...
-
-clean:
-	rm *.o
-
-run:
-	bash test.sh
-
-.IGNORE: run-error1 run-error2 run-error3
-
-cat-error1:
```
Comment (jmhossler) [inline makefile:30]: what happened to all the makefile test cases?

Comment (jmhossler) [reply makefile:30]: it was part of the requirements to have these as make commands

**Review by jmhossler — COMMENTED** [review]
if you do want to move the test cases into the test.sh, you should change the README to reflect that.

File: makefile (hunk 1 above)
Comment (pdietl) [reply makefile:30]: Class is over. I'll change the makefile.

Comment (jmhossler) [reply makefile:30]: okay, but update README to reflect the way to run the tests.

Comment (pdietl) [reply makefile:30]: README updated

**Pull request merged**